tests/Test.OneDriveSdk/ChunkedUploadProviderTests.cs

```diff
@@ -0,0 +1,263 @@
+﻿// ------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+using Microsoft.Graph;
+
+using Test.OneDrive.Sdk.Mocks;
+
+namespace Test.OneDrive.Sdk
+{
+
+    using Microsoft.OneDrive.Sdk;
+    using Microsoft.OneDrive.Sdk.Helpers;
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+    using Moq;
+
+    [TestClass]
+    public class ChunkedUploadProviderTests
```

Comment:
>ChunkedUploadProviderTests [](start = 17, length = 26)

I don't see any tests for the retry logic or the exception throwing during an upload